server/services/container.ts

```diff
@@ -0,0 +1,40 @@
+type FilterParams<T> = (value: T, index: number, array: T[]) => unknown
+type MapParams<T, R> = (t: T, index: number, array: T[]) => R
+type ExcludedItem = [item: unknown, reason: string]
```

Comment:
`item` is of unknown type as if you have a pipeline of operations, different steps could produce excluded items in different structures.